src/client/js/main.jsx

```diff
@@ -1,14 +1,140 @@
-var Test = React.createClass({
-    render() {
-        return (
-            <div>{this.props.msg}</div>
-        );
-    }
+/*
+
+	Copied, pasted and edited code from react-router example. To refine later
+
+ */
+
+
+var { Route, DefaultRoute, RouteHandler, Link } = ReactRouter;
+
+var App = React.createClass({
+  getInitialState: function () {
+    return { memes: getAllMemes() };
+  },
+
+  // App's render function
+  render: function () {
+    
+  	// Set up links to other memes
+    var links = this.state.memes.map(function (meme) {
+      
+      return (
+        <li key={meme.name}>
+        
+          <Link
+
+            to="memes-detail"
+
+            params={{ name: meme.name.toLowerCase().replace(/\s/g, "_") }}
+          >
+          	{meme.name}
+          </Link>
+        </li>
+      );
+    });
+
+    // Return app's view to render function
+    return (
+      <div className="App">
+        <div className="Detail">
+          <RouteHandler/>
+        </div>
+        <ul className="Master">
+          {links}
+        </ul>
+      </div>
+    );
+  }
+});
+
+
+/* Index */
```

Comment:
Let's make all these comments jsdoc comments so that we can easily parse them with jsdoc in the near future.